crates/lsp_test/src/lsp_client.rs

```diff
@@ -87,9 +88,22 @@ impl TestClient {
     pub async fn initialize(&mut self) -> i64 {
         let params: Option<lsp_types::InitializeParams> = std::mem::take(&mut self.init_params);
         let params = params.unwrap_or_default();
+        let params = Self::with_client_info(params);
         self.request::<lsp_types::request::Initialize>(params).await
     }
 
+    // Regardless of how we got the params, ensure the client name is set to
+    // `AirTestClient` so we can recognize it when we set up global logging.
+    fn with_client_info(
+        mut init_params: lsp_types::InitializeParams,
+    ) -> lsp_types::InitializeParams {
+        init_params.client_info = Some(ClientInfo {
+            name: String::from("AirTestClient"),
+            version: None,
+        });
+        init_params
+    }
```

Comment:
I like using `AirTestClient` as the way to determine "are we testing or not" rather than `cfg(debug_assertions)` because this will also "look like" testing mode in integration tests!